pipeline/runtime/run.go

```diff
@@ -110,12 +117,48 @@ func executeRunStep(ctx context.Context, f RunFunc, r *api.StartStepRequest, out
 					})
 				}
 			}
-			return exited, outputs, exportEnvs, artifact, outputsV2, string(optimizationState), err
 		} else if len(r.OutputVars) > 0 {
 			// only return err when output vars are expected
-			return exited, outputs, exportEnvs, artifact, nil, string(optimizationState), err
+			finalErr = err
+			for _, key := range r.OutputVars {
+				if _, ok := outputs[key]; ok {
+					output := &api.OutputV2{
+						Key:   key,
+						Value: outputs[key],
+						Type:  outputVariableTypeString,
+					}
+					outputsV2 = append(outputsV2, output)
+				}
+			}
+		} else {
+			for key, value := range outputs {
+				output := &api.OutputV2{
+					Key:   key,
+					Value: value,
+					Type:  outputVariableTypeString,
+				}
+				outputsV2 = append(outputsV2, output)
+			}
 		}
-		return exited, outputs, exportEnvs, artifact, nil, string(optimizationState), nil
+
+		//checking exported secrets from plugins if any
+		_, secretErr := os.Stat(outputSecretsFile)
```

Comment:
Let's log if `secretErr != nil`

Reply:
@vistaarjuneja 
Why do we need to log it ? this will return err for all the existing plugins as they dont use secret file as of now . this check is to understand whether a secret is exported or not . 
if we log it , the error will be logged for all the existing plugins .

Reply:
Ack, let's change it to this in a single line:
```
if _, err := os.Stat(outputSecretsFile); err == nil {
  
}
```